staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrole.go

```diff
@@ -147,7 +147,10 @@ func (c *CreateClusterRoleOptions) Validate() error {
 			}
 		}
 		if err := c.validateResource(); err != nil {
-			return err
+			if c.DryRunStrategy != cmdutil.DryRunClient {
+				return err
+			}
+			fmt.Fprintf(c.ErrOut, "Warning (Server side): '%v'\n", err.Error())
```

Comment:
Right now this is printing the error directly from [here](https://github.com/kubernetes/kubernetes/blob/84ef40e9524d95d968147a7757ebd2be8deae7fc/staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go#L323). To make it easier to understand for the user, we can sanitize with [this](https://github.com/kubernetes/kubectl/blob/c64f823bf35ad1a5fbe7110115ffc160ed8249ac/pkg/cmd/util/helpers.go#L195-L205).

Reply:
Why would you need to do that, and only skip client-side dry-run? If you return error that `checkErr` helper will be executed at a different level and will provide the information to the user, isn't that the goal?

Reply:
@soltysh The idea was to not pass through `checkErr` since that makes server side checks, but to just sanitize the error string based on the availability of resource group and version.

Reply:
The reason for skipping client side and not returning the error, is because if the resource is not available on server (which we figure out after looking at the [mapper](https://github.com/kubernetes/kubectl/blob/5ef32b94c02dc1206f4b5a3f00728363bb62e8dc/pkg/cmd/create/create_role.go#L325)) we need not error and instead just print warning since the user has specified it to specifically be a client-side dry run. The warning which we print here, could further be sanitized (if we need) by passing through the switch cases in `checkErr`.